### DISCUSSION TITLE
NHL long score delay

### DISCUSSION THREAD

**Discussion by hockey1491** [vasqued2/ha-teamtracker#190] [Q&A]
I recently set up team tracker to flash my govee lights when the Winnipeg Jets in the NHL score. Everything works but the delay on the goal being scored to the lights flashing is a couple of minutes which kind of defeats the purpose. I watched for the score update on ESPN and it also seemed to be a few minutes behind. Is this typical for ESPN and NHL scores? Or is there something I can do to address this?

**Comment by vasqued2** [comment]
I haven't watched an NHL game live in a while to see if this big of a delay is normal for it.  It isn't normal for other sports.  I sometimes get notified of NFL touchdowns before they actually happen on my TV.  I also use a streaming service which adds a bit of a delay to the video though.  Are you sure the delay is in the sensor and not somewhere in your automation?  2+ minutes seems excessive.  

Once the game starts, the API is called and the data is updated every 5 seconds.  If you check the Show Debug Info toggle, it will show the time of the last update on the card along with other info.  If there is a delay, it is on the ESPN side of things.

**Comment by vasqued2** [comment]
Caps/Flyers on ESPN is pretty close to the live video on YouTubeTV.  Looks like the API updates data every ~20 seconds.

**Comment by mrplow** [comment]
Sorry to hop on this issue, but I'm also finding the ESPN API quite slow to update NHL scores as well (go Canucks).

I created a sensor out of the opponent_score and team_score attributes so I could get some stats. I also did the same thing with the NHL API integration.

The NHL API integration is, on average, 30 seconds faster. With the highest difference of 46 seconds and the lowest of 11 seconds. This is just with minimal data though, only two games.

I'm guessing there isn't anything that can be done here since it is just ESPN being slow on their API updates.

**Comment by vasqued2** [comment]
It's all dependent on the ESPN API.  The sensor polls the API every 5 seconds while the game is in progress.  At most, there is a 5 second delay between when the ESPN API publishes the data and when the sensor gets it.

Nothing I can do on the sensor side.
